compiler/rustc_trait_selection/src/traits/select/candidate_assembly.rs

```diff
@@ -203,6 +203,18 @@ impl<'cx, 'tcx> SelectionContext<'cx, 'tcx> {
             }
         }
 
+        if self.tcx().features().impl_unified_exhaustive_const_traits && candidates.is_empty() {
+            let mut out = vec![];
+            self.assemble_candidates_from_unified_impls(stack.obligation, &mut out);
+            if !out.is_empty() {
+                assert!(out.len() == 1);
+                candidates.push(EvaluatedCandidate {
+                    candidate: out.pop().unwrap(),
+                    evaluation: crate::traits::EvaluationResult::EvaluatedToOk,
+                });
+            }
+        }
```

Comment:
I was hoping we could insert it unconditionally if the candidates list is empty after https://github.com/rust-lang/rust/blob/dd12cd6dc631b5e964d541d370ca863c2242376c/compiler/rustc_trait_selection/src/traits/select/candidate_assembly.rs#L329-L333

Did you try that?